titiler/endpoints/factory.py

```diff
@@ -878,11 +883,11 @@ def tile(
                 )
             timings.append(("format", round(t.elapsed * 1000, 2)))
 
-            headers["Server-Timing"] = ", ".join(
-                [f"{name};dur={time}" for (name, time) in timings]
-            )
-
-            headers["X-Assets"] = ",".join(data.assets)
+            if self.debug:
+                headers["Server-Timing"] = ", ".join(
+                    [f"{name};dur={time}" for (name, time) in timings]
+                )
+                headers["X-Assets"] = ",".join(data.assets)
```

Comment:
I wouldn't necessarily agree that `X-Assets` should be considered a debugging header. `Server-Timing` would never be used on the frontend, but `X-Assets` could actually be potentially used in the frontend to provide information about what images went into the tile.